examples/tags.rs

```diff
@@ -38,36 +38,57 @@ fn main() {
 }
 
 fn run(host: &str, user: Option<String>, passwd: Option<String>, image: &str) -> Result<()> {
-    let mut tcore = try!(Core::new());
-    let mut dclient = try!(
-        dkregistry::v2::Client::configure(&tcore.handle())
-            .registry(host)
-            .insecure_registry(false)
-            .username(user)
-            .password(passwd)
-            .build()
-    );
+    let mut tcore = Core::new()?;
+    let client = dkregistry::v2::Client::configure(&tcore.handle())
+        .registry(host)
+        .insecure_registry(false)
+        .username(user)
+        .password(passwd)
+        .build()?;
 
-    let futcheck = try!(dclient.is_v2_supported());
-    let supported = try!(tcore.run(futcheck));
-    if !supported {
-        return Err("API v2 not supported".into());
-    }
-
-    let fut_token = try!(dclient.login(&[&format!("repository:{}:pull", image)]));
-    let token_auth = try!(tcore.run(fut_token));
+    let futures = futures::future::ok::<_, dkregistry::errors::Error>(client)
+        .and_then(|dclient| {
+            dclient.is_v2_supported().and_then(|v2_supported| {
+                if !v2_supported {
+                    Err("API v2 not supported".into())
+                } else {
+                    Ok(dclient)
+                }
+            })
+        }).and_then(|dclient| {
+            dclient.is_auth(None).and_then(|is_auth| {
+                if is_auth {
+                    Err("no login performed, but already authenticated".into())
+                } else {
+                    Ok(dclient)
+                }
+            })
+        }).and_then(|dclient| {
+            dclient
+                .login(&[&format!("repository:{}:pull", image)])
```

Comment:
I would extract the scope-formatting outside (before) the start of the chain, so that it's harder for people to miss (here and everywhere else too).